src/main/java/com/epam/test_generator/controllers/LoginController.java

```diff
@@ -24,11 +29,22 @@
     private LoginService loginService;
 
     @PostMapping("/login")
-    public ResponseEntity<TokenDTO> login(@RequestBody @Valid LoginUserDTO userDTO, HttpServletRequest request)
-        throws Exception {
+    public ResponseEntity<TokenDTO> login(@RequestBody @Valid LoginUserDTO userDTO,
+                                          HttpServletRequest request) {
         loginService.checkPassword(userDTO, request);
         final String token = loginService.getLoginJWTToken(userDTO);
 
         return new ResponseEntity<>(new TokenDTO(token), HttpStatus.OK);
     }
+
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_LEAD", "ROLE_TEST_ENGINEER", "ROLE_GUEST"})
+    @ApiImplicitParam(name = "Authorization", value = "add here your token",
+        paramType = "header", dataType = "string", required = true)
+    @GetMapping("/refresh-token")
+    public ResponseEntity refreshToken(Authentication authentication) {
```

Comment:
IMO it is bad practice to return valid token if previous isn't expired. It encourages client to call this method whenever it wants, and may produce unlimited calls.

Line 46 may be written as follows

    .refreshToken(((AuthenticatedUser) authentication).getToken());

Reply:
done